tests/core/vm/test_sharding_vm.py

```diff
@@ -7,26 +12,117 @@
     new_sharding_transaction,
 )
 from tests.core.vm.contract_fixture import (
-    contract_bytecode,
-    contract_address,
+    simple_transfer_contract_bytecode,
+    simple_transfer_contract_address,
+    simple_contract_factory_bytecode,
+    CREATE2_contract_bytecode,
+    CREATE2_contract_address,
 )
 
 
-def test_sharding_transaction(shard_chain_without_block_validation):  # noqa: F811
+def test_sharding_apply_transaction(shard_chain_without_block_validation):  # noqa: F811
     chain = shard_chain_without_block_validation
-    deploy_tx = new_sharding_transaction(contract_address, b'', 0, b'', b'', contract_bytecode)
+    # First test: simple ether transfer contract
+    first_deploy_tx = new_sharding_transaction(
+        simple_transfer_contract_address,
+        b'',
+        0,
+        b'',
+        b'',
+        simple_transfer_contract_bytecode
+    )
 
     vm = chain.get_vm()
-    computation = vm.apply_transaction(deploy_tx)
+    computation = vm.apply_transaction(first_deploy_tx)
     assert not computation.is_error
+    gas_used = vm.block.header.gas_used
+    assert gas_used > first_deploy_tx.intrinsic_gas
+    last_gas_used = gas_used
 
     # Transfer ether to recipient
     recipient = decode_hex('0xa94f5374fce5edbc8e2a8697c15331677e6ebf0c')
     amount = 100
-    tx_initiator = contract_address
+    tx_initiator = simple_transfer_contract_address
     transfer_tx = new_sharding_transaction(tx_initiator, recipient, amount, b'', b'', b'')
 
-    computation = vm.execute_transaction(transfer_tx)
+    computation = vm.apply_transaction(transfer_tx)
     assert not computation.is_error
+    gas_used = vm.block.header.gas_used - last_gas_used
+    assert gas_used > transfer_tx.intrinsic_gas
+    last_gas_used = vm.block.header.gas_used
     with vm.state.state_db(read_only=True) as state_db:
         assert state_db.get_balance(recipient) == amount
+
+    # Second test: contract that deploy new contract with CREATE2
+    second_deploy_tx = new_sharding_transaction(
+        CREATE2_contract_address,
+        b'',
```

Comment:
Can you change this to use keyword arguments so that it's clear what all of these similar parameters are.